TypeError when calling wait_for_storage_pods

```
Error Message

TypeError: argument of type 'NoneType' is not iterable

Stacktrace

multiregion_mirror_setup_session = (<ocs_ci.ocs.resources.objectbucket.MCGS3Bucket object at 0x7f294c1396d8>, [<ocs_ci.ocs.resources.backingstore.BackingStore object at 0x7f294c28da20>, <ocs_ci.ocs.resources.backingstore.BackingStore object at 0x7f2947aaf390>])

    @skipif_aws_creds_are_missing
    @post_upgrade
    @pytest.mark.polarion_id("OCS-2220")
    def test_storage_pods_running(multiregion_mirror_setup_session):
        """
        Test that all pods from openshift-storage namespace have status Running
        or Completed after upgrade is completed.
    
        multiregion_mirror_setup_session fixture is present during this test to
        make sure that NooBaa backing stores from other upgrade tests were
        not yet deleted. This is done to test scenario from BZ 1823775.
    
        """
>       wait_for_storage_pods(timeout=10), 'Some pods were not in expected state'

tests/ecosystem/upgrade/test_resources.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ocs_ci/ocs/resources/pod.py:1311: in wait_for_storage_pods
    pod for pod in all_pod_obj if constants.ROOK_CEPH_DETECT_VERSION_LABEL not in pod.get_labels()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <list_iterator object at 0x7f294e5d6d30>

>       pod for pod in all_pod_obj if constants.ROOK_CEPH_DETECT_VERSION_LABEL not in pod.get_labels()
    ]
E   TypeError: argument of type 'NoneType' is not iterable

ocs_ci/ocs/resources/pod.py:1311: TypeError
```

There is a problem introduced in PR https://github.com/red-hat-storage/ocs-ci/pull/3213. There is a check for all pods in openshift-storage namespace if they don't have ROOK_CEPH_DETECT_VERSION_LABEL label.
Some pods (like compute-0-debug and compute-1-debug) don't have labels defined in their yaml definition. Function get_labels(): https://github.com/red-hat-storage/ocs-ci/blob/c4d5cd540436e09ee31dd809b24d2d0d57ef6f7d/ocs_ci/ocs/resources/pod.py#L189 uses get() on items in yaml and if the key is not found then None is returned (https://www.w3schools.com/python/ref_dictionary_get.asp). This causes an error when `constants.ROOK_CEPH_DETECT_VERSION_LABEL not in pod.get_labels()` is performed.

same issue:
        for pod in all_pod_obj
>       if constants.ROOK_CEPH_DETECT_VERSION_LABEL not in pod.get_labels()
    ]
E   TypeError: argument of type 'NoneType' is not iterabl

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -1342,10 +1342,12 @@ def wait_for_storage_pods(timeout=200):
     """
     all_pod_obj = get_all_pods(namespace=defaults.ROOK_CLUSTER_NAMESPACE)
     # Ignoring pods with "app=rook-ceph-detect-version" app label
+
     all_pod_obj = [
         pod
         for pod in all_pod_obj
-        if constants.ROOK_CEPH_DETECT_VERSION_LABEL not in pod.get_labels()
+        if pod.get_labels()
+        and constants.ROOK_CEPH_DETECT_VERSION_LABEL not in pod.get_labels()
     ]
 
     for pod_obj in all_pod_obj:

